### ISSUE TITLE
Cash Account Balance Not Included in Net Worth

### ISSUE THREAD

**Issue by maureenferreira** [firefly-iii/firefly-iii#9047]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

The balance in my Cash Account isn't factored into my Net Worth, and I've made sure "Include in Net Worth" is enabled in the settings for that account. Further, when I make a withdrawal from my Cash Account, my Net Worth still isn't updated.

### Debug information

Debug information generated at 2024-07-14 16:32:04 for Firefly III version **v6.1.18**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.18 / v2.1.0 / 24 (exp. 24)</td></tr><tr><td>PHP version</td><td>8.3.8 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>1104, base <span>#</span>85</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Amsterdam + Atlantic/Reykjavik</td></tr><tr><td>App environment</td><td>production, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td></td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td></td></tr><tr><td>Session start</td><td>2024-07-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-07-31 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; rv:128.0) Gecko/20100101 Firefox/128.0</td></tr></tbody></table>

### Expected behaviour

I expect the balance of my Cash Account to be factored into my Net Worth.

### Steps to reproduce

1. Ensure "Include in Net Worth" is enabled in the settings for your Cash Account
2. Add a non-zero balance to your Cash Account
3. Check your Net Worth

**Comment by JC5** [comment]
Hey, thanks for opening an issue.

I can't replicate this on the demo site. It works as expected. 

Try browsing to `/flush`, data may be cached.

**Comment by maureenferreira** [comment]
> I can't replicate this on the demo site. It works as expected.
I tried replicating it on the demo site and had the same exact issue, so that's really weird.
> Try browsing to `/flush`, data may be cached.
Just tried it, it didn't work.

**Comment by JC5** [comment]
I add 2000 to the cash account and I see the net worth rising with 2000. What did you try?

**Comment by maureenferreira** [comment]
I just added a cash account, enabled 'Include in Net Worth', added some money to it (I forgot how much right now), and checked to see if the Net Worth changed,

**Comment by maureenferreira** [comment]
So, I just added another account and the same thing happened. This time, it's just a Default asset account.

**Comment by JC5** [comment]
OK which net worth are you talking about, exactly?

**Comment by maureenferreira** [comment]
The big one on the Dashboard page.
![image](https://github.com/user-attachments/assets/5038d0e0-1678-4d1d-ac83-f6ec1e8ceb6b)

**Comment by JC5** [comment]
Sorry, it works for me, it really does.

- Create asset account. Create deposit. See net worth rise. 
- Create asset account. Create withdrawal. See net worth drop. 
- Create asset account. Set opening balance. See net worth change.

Same with cash account type.

**Comment by maureenferreira** [comment]
That's... bizarre. Is there anything else I can do to troubleshoot this and maybe figure out the issue?

**Comment by JC5** [comment]
You're not transferring the money from another account, right?

**Comment by maureenferreira** [comment]
Nope.

**Comment by maureenferreira** [comment]
Okay, I just figured out what's going on. Something you said stood out to me, "Opening balance." I was setting the virtual balance for my other accounts since they're things like gift cards so I figured it didn't make sense to set a date for an opening balance, and I figured virtual balance would basically do the same thing.

What's the different between the virtual balance and opening balance and why isn't the virtual balance added to your Net Worth?
